Expose an option to avoid notifying testers of new releases

cysp opened the pull request:
When distributing a release, it is now possible to choose not to notify testers.

Closes #602

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/appcenter-cli?pullRequest=643) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/appcenter-cli?pullRequest=643) before we can accept your contribution.<br/><br/>:x: cysp [sign now](https://cla.opensource.microsoft.com/microsoft/appcenter-cli?pullRequest=643)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/appcenter-cli?pullRequest=643) it.</sub>

cysp commented on src/commands/distribute/release.ts:
```diff
@@ -284,13 +289,13 @@ export default class ReleaseBinaryCommand extends AppCommand {
     });
     await addGroupToRelease({
-      client, releaseId, distributionGroup: distributionGroupResponse, app: this.app, destination: this.distributionGroup, destinationType: "group", mandatory: false, silent: false
+      client, releaseId, distributionGroup: distributionGroupResponse, app: this.app, destination: this.distributionGroup, destinationType: "group", mandatory: false, silent: !notifyTesters
```
This `silent` becomes `notifyTesters` again on the other side of this function call. Would you like this option name to change to maintain consistency?
https://github.com/microsoft/appcenter-cli/blob/master/src/commands/distribute/lib/distribute-util.ts#L52

jpeg-ms commented:
I believe the option should be called `silent` to be consistent with the existing option for the command `distribute releases add-destination --silent`.

cysp commented:
Sure, I've renamed the option to `--silent` (though `--no-silent` reads a bit funny now 🙃)

I saw incorrect behaviour – with the command line flag seemingly having no effect – until I removed the `@defaultValue("false")` annotation from the `silent` property, which I don't understand?

<details>
<summary>diff:</summary>

```diff
diff --git c/src/commands/distribute/release.ts w/src/commands/distribute/release.ts
index eb4c4fd..934784e 100644
--- c/src/commands/distribute/release.ts
+++ w/src/commands/distribute/release.ts
@@ -1,4 +1,4 @@
-import { AppCommand, CommandResult, ErrorCodes, defaultValue, failure, hasArg, help, longName, required, shortName, success } from "../../util/commandline";
+import { AppCommand, CommandResult, ErrorCodes, failure, hasArg, help, longName, required, shortName, success } from "../../util/commandline";
 import { AppCenterClient, models, clientRequest, ClientResponse } from "../../util/apis";
 import { out } from "../../util/interaction";
 import { inspect } from "util";
@@ -45,10 +45,9 @@ export default class ReleaseBinaryCommand extends AppCommand {
   @hasArg
   public releaseNotesFile: string;
 
-  @help("Notify testers of this release")
-  @longName("notify-testers")
-  @defaultValue("true")
-  public notifyTesters: boolean;
+  @help("Do not notify testers of this release")
+  @longName("silent")
+  public silent: boolean;
 
   public async run(client: AppCenterClient): Promise<CommandResult> {
     const app: DefaultApp = this.app;
@@ -87,7 +86,7 @@ export default class ReleaseBinaryCommand extends AppCommand {
     const releaseId = this.extractReleaseId(releaseUrl);
 
     debug("Distributing the release");
-    await this.distributeRelease(client, app, releaseId, releaseNotesString, this.notifyTesters);
+    await this.distributeRelease(client, app, releaseId, releaseNotesString, this.silent);
 
     debug("Retrieving the release");
     const releaseDetails = await this.getDistributeRelease(client, app, releaseId);
@@ -289,13 +288,13 @@ export default class ReleaseBinaryCommand extends AppCommand {
     }
   }
 
-  private async distributeRelease(client: AppCenterClient, app: DefaultApp, releaseId: number, releaseNotesString: string, notifyTesters: boolean): Promise<void> {
+  private async distributeRelease(client: AppCenterClient, app: DefaultApp, releaseId: number, releaseNotesString: string, silent: boolean): Promise<void> {
     await this.putReleaseDetails(client, app, releaseId, releaseNotesString);
     const distributionGroupResponse = await getDistributionGroup({
       client, releaseId, app: this.app, destination: this.distributionGroup, destinationType: "group"
     });
     await addGroupToRelease({
-      client, releaseId, distributionGroup: distributionGroupResponse, app: this.app, destination: this.distributionGroup, destinationType: "group", mandatory: false, silent: !notifyTesters
+      client, releaseId, distributionGroup: distributionGroupResponse, app: this.app, destination: this.distributionGroup, destinationType: "group", mandatory: false, silent: silent
     });
   }
 }
diff --git c/test/commands/distribute/release/release-test.ts w/test/commands/distribute/release/release-test.ts
index 96bb323..6219b94 100644
--- c/test/commands/distribute/release/release-test.ts
+++ w/test/commands/distribute/release/release-test.ts
@@ -179,7 +179,7 @@ describe("release command", () => {
 
   });
 
-  context("notify-testers", () => {
+  context("silent", () => {
     describe("when notifying testers by default", () => {
       beforeEach(() => {
         expectedRequestsScope = setupSuccessfulGetDistributionGroupUsersResponse(
@@ -189,7 +189,7 @@ describe("release command", () => {
                 setupSuccessfulCreateReleaseResponse(
                   setupSuccessfulAddGroupResponse(
                     setupSuccsessFulGetDistributionGroupResponse(
-                      Nock(fakeHost)), true))))));
+                      Nock(fakeHost))))))));
         skippedRequestsScope = setupSuccessfulAbortUploadResponse(Nock(fakeHost));
       });
 
@@ -216,7 +216,7 @@ describe("release command", () => {
                 setupSuccessfulCreateReleaseResponse(
                   setupSuccessfulAddGroupResponse(
                     setupSuccsessFulGetDistributionGroupResponse(
-                      Nock(fakeHost)), true))))));
+                      Nock(fakeHost))))))));
         skippedRequestsScope = setupSuccessfulAbortUploadResponse(Nock(fakeHost));
       });
 
@@ -225,7 +225,7 @@ describe("release command", () => {
         const releaseFilePath = createFile(tmpFolderPath, releaseFileName, releaseFileContent);
 
         // Act
-        const command = new ReleaseBinaryCommand(getCommandArgs(["-f", releaseFilePath, "-r", releaseNotes, "-g", fakeDistributionGroupName, "--notify-testers"]));
+        const command = new ReleaseBinaryCommand(getCommandArgs(["-f", releaseFilePath, "-r", releaseNotes, "-g", fakeDistributionGroupName, "--no-silent"]));
         const result = await command.execute();
 
         // Assert
@@ -243,7 +243,7 @@ describe("release command", () => {
                 setupSuccessfulCreateReleaseResponse(
                   setupSuccessfulAddGroupResponse(
                     setupSuccsessFulGetDistributionGroupResponse(
-                      Nock(fakeHost)), false))))));
+                      Nock(fakeHost)), true))))));
         skippedRequestsScope = setupSuccessfulAbortUploadResponse(Nock(fakeHost));
       });
 
@@ -252,7 +252,7 @@ describe("release command", () => {
         const releaseFilePath = createFile(tmpFolderPath, releaseFileName, releaseFileContent);
 
         // Act
-        const command = new ReleaseBinaryCommand(getCommandArgs(["-f", releaseFilePath, "-r", releaseNotes, "-g", fakeDistributionGroupName, "--no-notify-testers"]));
+        const command = new ReleaseBinaryCommand(getCommandArgs(["-f", releaseFilePath, "-r", releaseNotes, "-g", fakeDistributionGroupName, "--silent"]));
         const result = await command.execute();
 
         // Assert
@@ -481,28 +481,28 @@ describe("release command", () => {
     }).reply(404);
   }
 
-  function setupSuccessfulAddGroupResponse(nockScope: Nock.Scope, notifyTesters = true): Nock.Scope {
+  function setupSuccessfulAddGroupResponse(nockScope: Nock.Scope, silent = false): Nock.Scope {
     const postAddReleaseGroupDestinationUrl = `/v0.1/apps/${fakeAppOwner}/${fakeAppName}/releases/${fakeReleaseId}/groups`;
     const expectedBody = {
       id: fakeGroupId,
       mandatory_update: false,
-      notify_testers: notifyTesters
+      notify_testers: !silent
     };
 
     return nockScope.post(postAddReleaseGroupDestinationUrl, expectedBody)
     .reply(201, {
       id: fakeGroupId,
       mandatory_update: false,
-      notify_testers: notifyTesters
+      notify_testers: !silent
     });
   }
 
-  function setupFailedAddGroupResponse(nockScope: Nock.Scope, notifyTesters = true): Nock.Scope {
+  function setupFailedAddGroupResponse(nockScope: Nock.Scope, silent = false): Nock.Scope {
     const postAddReleaseGroupDestinationUrl = `/v0.1/apps/${fakeAppOwner}/${fakeAppName}/releases/${fakeReleaseId}/groups`;
     const expectedBody = {
       id: fakeGroupId,
       mandatory_update: false,
-      notify_testers: notifyTesters
+      notify_testers: !silent
     };
 
     return nockScope.post(postAddReleaseGroupDestinationUrl, expectedBody)
```
</details>
